### PR TITLE
Merge CosmosAsyncContainerResponse and CosmosContainerResponse

### PR DESCRIPTION
**Pull request by xinlian12**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/examples/HelloWorldDemo.java [hunk 1]
```diff
@@ -4,11 +4,15 @@
 
 import com.azure.cosmos.CosmosAsyncClient;
 import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
 import com.azure.cosmos.CosmosClientBuilder;
 import com.azure.cosmos.models.PartitionKey;
 import reactor.core.publisher.Mono;
 
 public class HelloWorldDemo {
+    private static String DATABASE_NAME = "contoso-travel";
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/examples/HelloWorldDemo.java:13]:
```suggestion
    private static final String DATABASE_NAME = "contoso-travel";
```

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/examples/HelloWorldDemo.java]: changed

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java [hunk 2]
```diff
@@ -256,15 +256,17 @@ private void createClients() {
                     }
 
                     try {
-                        cosmosAsyncContainer = cosmosAsyncDatabase.getContainer(this.configuration.getCollectionId()).read().block().getContainer();
+                        cosmosAsyncDatabase.getContainer(this.configuration.getCollectionId()).read().block();
+                        cosmosAsyncContainer = cosmosAsyncDatabase.getContainer(this.configuration.getCollectionId());
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java:259]: Good point, changed

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/SyncBenchmark.java [hunk 3]
```diff
@@ -131,13 +131,14 @@ public T apply(T o, Throwable throwable) {
         }
 
         try {
-            cosmosContainer = cosmosDatabase.getContainer(this.configuration.getCollectionId()).read().getContainer();
+            cosmosDatabase.getContainer(this.configuration.getCollectionId()).read();
+            cosmosContainer = cosmosDatabase.getContainer(this.configuration.getCollectionId());
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/SyncBenchmark.java:134]: Changed

File: sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/examples/HelloWorldDemo.java [hunk 4]
```diff
@@ -4,11 +4,15 @@
...
+    private static String CONTAINER_NAME = "passengers";
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/examples/HelloWorldDemo.java]: Changed.

**Comment by xinlian12** [comment]
Run the following tests and passed:
CosmosContainerContentResponseOnWriteTest
CosmosContainerTest
CollectionQueryTest
CollectionCrudTest

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java [hunk 5]
```diff
@@ -99,18 +99,22 @@
...
         try {
-            cosmosAsyncContainer = cosmosAsyncDatabase.getContainer(
+            cosmosAsyncDatabase.getContainer(
                 this.configuration.getCollectionId()
             ).read().doOnError(error ->
                 logger.error("Database {} creation failed due to ", this.configuration.getDatabaseId(), error)
-            ).block().getContainer();
+            ).block();
+
+            cosmosAsyncContainer = cosmosAsyncDatabase.getContainer(this.configuration.getCollectionId());
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java]: changed.

**Pull request merged**